pkg/webhook/mutatingwebhook.go

```diff
@@ -109,8 +109,25 @@ func (si *SidecarInjector) Handle(_ context.Context, req admission.Request) admi
 
 	klog.Infof("mutating Pod: Name %q, GenerateName %q, Namespace %q, CPU limit %q, memory limit %q, ephemeral storage limit %q", pod.Name, pod.GenerateName, pod.Namespace, configCopy.CPULimit.String(), configCopy.MemoryLimit.String(), configCopy.EphemeralStorageLimit.String())
 	// the gcsfuse sidecar container has to before the containers that consume the gcsfuse volume
-	pod.Spec.Containers = append([]corev1.Container{GetSidecarContainerSpec(configCopy)}, pod.Spec.Containers...)
-	pod.Spec.Volumes = append([]corev1.Volume{GetSidecarContainerVolumeSpec()}, pod.Spec.Volumes...)
+	useExperimentalLocalFileCache := false
+	hasCacheVolume := false
+	for _, v := range pod.Spec.Volumes {
+		if v.Name == CacheVolumeName {
+			hasCacheVolume = false
+		}
+		if v.CSI == nil || v.CSI.VolumeAttributes == nil {
```

Comment:
Does this mean for the read cache test, we only focus on the CSI inline volume?

I guess for PV/PVC usage, we will rely on the Pod annotations in the final design, is that correct?

Reply:
If the answers are yes, could you add comments as well? Or a TODO to remove this logic in the future?

Reply:
Resolving as discussed offline. This feature no longer necessitates driver changes.